src/state/ppangoChef/hooks.ts

```diff
@@ -349,3 +358,64 @@ export function usePangoChefInfos() {
     pairs,
   ]);
 }
+
+export function useUserAPR(stakingInfo?: PangoChefInfo) {
+  const blockTime = useGetBlockTimestamp();
+
+  return useMemo(() => {
+    if (!stakingInfo) return '0';
+
+    const userBalance = stakingInfo.userValueVariables.balance;
+    const userSumOfEntryTimes = stakingInfo.userValueVariables.sumOfEntryTimes;
+
+    const poolBalance = stakingInfo.valueVariables.balance;
+    const poolSumOfEntryTimes = stakingInfo.valueVariables.sumOfEntryTimes;
+
+    if (userBalance.isZero() || poolBalance.isZero() || !blockTime) return '0';
+
+    //userAPR = poolAPR * (blockTime - (userValueVariables.sumOfEntryTimes / userValueVariables.balance)) / (blockTime - (poolValueVariables.sumOfEntryTimes / poolValueVariables.balance))
+    const a = userSumOfEntryTimes.div(userBalance);
+    const b = poolSumOfEntryTimes.div(poolBalance);
+    const c = blockTime.sub(a);
+    const d = blockTime.sub(b);
+    return BigNumber.from(stakingInfo.stakingApr ?? 0)
+      .mul(c)
+      .div(d)
+      .toString();
+  }, [blockTime, stakingInfo]);
+}
+
+export function useUserRewardRate(stakingInfo: PangoChefInfo) {
```

Comment:
lets rename this hook to indicate its PangoChef reward rate something like `useUserPangoChefRewardRate`